npm/pkg/dataplane/dataplane.go

```diff
@@ -149,6 +174,9 @@ func (dp *DataPlane) AddToSets(setNames []*ipsets.IPSetMetadata, podMetadata *Po
 			klog.Infof("[DataPlane] {AddToSet} pod key %s not found in updatePodCache. creating a new obj", podMetadata.PodKey)
 			updatePod = newUpdateNPMPod(podMetadata)
 			dp.updatePodCache.cache[podMetadata.PodKey] = updatePod
+
+			// add to queue only if not in the cache/queue already
+			dp.updatePodCache.queue = append(dp.updatePodCache.queue, podMetadata.PodKey)
```

Comment:
Where are we checking if the key is not in queue ? we will need to have a 2nd check here,